src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -319,10 +319,22 @@ def update_container_settings(resource_group_name, name, docker_registry_server_
     settings = []
     if docker_registry_server_url is not None:
         settings.append('DOCKER_REGISTRY_SERVER_URL=' + docker_registry_server_url)
+
+    if (not docker_registry_server_user and not docker_registry_server_password and
+            docker_registry_server_url and '.azurecr.io' in docker_registry_server_url):
```

Comment:
The sniffing mechanism on `.azurecr.io` was communicated to ACR team. As ACR has no support in sovereign clouds, we are fine for now. I will open a tracking bug to get it into CLI cloud context when there are ready.